src/saltext/vmware/states/vmc_nat_rules.py

```diff
@@ -0,0 +1,512 @@
+        "logging": logging,
+        "sequence_number": sequence_number,
+        "display_name": display_name,
+    }
+
+    input_dict = {k: v for k, v in input_dict.items() if v != vmc_constants.VMC_NONE}
+
+    get_nat_rule = __salt__["vmc_nat_rules.get_by_id"](
+        hostname=hostname,
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        tier1=tier1,
+        nat=nat,
+        nat_rule=nat_rule,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+
+    existing_nat_rule = None
+
+    if "error" not in get_nat_rule:
+        log.info("Nat rule found with Id %s", nat_rule)
+        existing_nat_rule = get_nat_rule
+    elif NAT_RULE_NOT_FOUND_ERROR not in get_nat_rule["error"]:
+        return vmc_state._create_state_response(
+            name=name, comment=get_nat_rule["error"], result=False
+        )
+
+    if __opts__.get("test"):
+        log.info("present is called with test option")
+        if existing_nat_rule:
+            return vmc_state._create_state_response(
+                name=name, comment="State present will update nat rule {}".format(nat_rule)
+            )
+        else:
+            return vmc_state._create_state_response(
+                name=name, comment="State present will create nat rule {}".format(nat_rule)
+            )
+
+    if existing_nat_rule:
+        updatable_keys = input_dict.keys()
+        is_update_required = vmc_state._check_for_updates(
+            existing_nat_rule, input_dict, updatable_keys, ["translated_ports", "tags"]
+        )
+
+        if is_update_required:
+            updated_nat_rule = __salt__["vmc_nat_rules.update"](
+                hostname=hostname,
+                refresh_key=refresh_key,
+                authorization_host=authorization_host,
+                org_id=org_id,
+                sddc_id=sddc_id,
+                tier1=tier1,
+                nat=nat,
+                nat_rule=nat_rule,
+                verify_ssl=verify_ssl,
+                cert=cert,
+                action=action,
+                destination_network=destination_network,
+                source_network=source_network,
+                translated_network=translated_network,
+                translated_ports=translated_ports,
+                scope=scope,
+                service=service,
+                enabled=enabled,
+                firewall_match=firewall_match,
+                logging=logging,
+                description=description,
+                tags=tags,
+                sequence_number=sequence_number,
+                display_name=display_name,
+            )
+
+            if "error" in updated_nat_rule:
+                return vmc_state._create_state_response(
+                    name=name, comment=updated_nat_rule["error"], result=False
+                )
+
+            get_nat_rule_after_update = __salt__["vmc_nat_rules.get_by_id"](
```

Comment:
**question** why get after update? Does `updated_nat_rule` not contain the new information?